src/main/java/com/krishagni/openspecimen/msk2/importer/CarsImporter.java

```diff
@@ -0,0 +1,278 @@
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Configurable;
+
+import com.krishagni.catissueplus.core.administrative.events.ListPvCriteria;
+import com.krishagni.catissueplus.core.administrative.events.PvDetail;
+import com.krishagni.catissueplus.core.biospecimen.domain.CollectionProtocol;
+import com.krishagni.catissueplus.core.biospecimen.domain.CollectionProtocolEvent;
+import com.krishagni.catissueplus.core.biospecimen.domain.SpecimenRequirement;
+import com.krishagni.catissueplus.core.biospecimen.events.CollectionProtocolDetail;
+import com.krishagni.catissueplus.core.biospecimen.events.CollectionProtocolEventDetail;
+import com.krishagni.catissueplus.core.biospecimen.events.CollectionProtocolSiteDetail;
+import com.krishagni.catissueplus.core.biospecimen.events.SpecimenRequirementDetail;
+import com.krishagni.catissueplus.core.biospecimen.repository.DaoFactory;
+import com.krishagni.catissueplus.core.biospecimen.services.CollectionProtocolService;
+import com.krishagni.catissueplus.core.common.PlusTransactional;
+import com.krishagni.catissueplus.core.common.errors.OpenSpecimenException;
+import com.krishagni.catissueplus.core.common.events.RequestEvent;
+import com.krishagni.catissueplus.core.common.events.ResponseEvent;
+import com.krishagni.catissueplus.core.common.events.UserSummary;
+import com.krishagni.catissueplus.core.common.service.PermissibleValueService;
+import com.krishagni.catissueplus.core.importer.events.ImportObjectDetail;
+import com.krishagni.catissueplus.core.importer.services.ObjectImporter;
+
+@Configurable
+public class CarsImporter implements ObjectImporter<CarsDetail, CarsDetail> {
+	
+	@Autowired
+	private CollectionProtocolService cpSvc;
+	
+	@Autowired
+	private PermissibleValueService pvSvc;
+	
+	@Autowired
+	private DaoFactory daoFactory;
+	
+	private Map<String, CollectionProtocolDetail> cpMap = new HashMap<>();
+	
+	private Map<String, CollectionProtocolEventDetail> eventsMap = new HashMap<>();
+	
+	private Map<String, SpecimenRequirementDetail> srMap = new HashMap<>();
+	
+	@Override
+	public ResponseEvent<CarsDetail> importObject(RequestEvent<ImportObjectDetail<CarsDetail>> req) {
+		try {
+			ImportObjectDetail<CarsDetail> detail = req.getPayload();
+			importRecord(detail);
+			return ResponseEvent.response(detail.getObject());
+		} catch (OpenSpecimenException ose) {
+			return ResponseEvent.error(ose);
+		} catch (Exception e) {
+			return ResponseEvent.serverError(e);
+		}
+	}
+	
+	@PlusTransactional
+	private ResponseEvent<Object> importRecord(ImportObjectDetail<CarsDetail> detail) throws Exception {
+		CollectionProtocolEventDetail event = toEvent(detail.getObject());
```

Comment:
I would do this as rather than overloading a single function with multiple responsibilities

CollectionProtocolEventDetail event = toEvent(detail.getObject());
SpecimenRequirementDetail  sr = toSr(detail.getObject());